lib/Zonemaster/Backend/Config.pm

```diff
@@ -15,27 +15,52 @@ use Readonly;
 use Zonemaster::Backend::Validator qw( :untaint );
 use Zonemaster::Backend::DB;
 
-our $path;
+Readonly my @SIG_NAME => split ' ', $Config{sig_name};
 
-if ($ENV{ZONEMASTER_BACKEND_CONFIG_FILE}) {
-    $path = $ENV{ZONEMASTER_BACKEND_CONFIG_FILE};
-}
-else {
-    my @search_paths = (
-        '/etc/zonemaster/backend_config.ini',
-        '/usr/local/etc/zonemaster/backend_config.ini',
-        dist_file('Zonemaster-Backend', "backend_config.ini")
-    );
+=head1 STATIC METHODS
 
-    for my $default_path (@search_paths) {
-        if ( -e $default_path ) {
-            $path = $default_path;
-            last;
-        }
-    }
-}
+=head2 get_default_path
 
-Readonly my @SIG_NAME => split ' ', $Config{sig_name};
+Determine the path for the default backend_config.ini file.
+A list of values and locations are checked and the first match is returned.
+If all places are checked and no file is found, an exception is thrown.
+
+This procedure is idempotent - i.e. if you call this procedure multiple times
+the same value is returned no matter if envirnmental circumstances have changed.
```

Comment:
I do not understand this. Please explain.

Say that the environment variable is unset, but /etc/zonemaster/backend_config.ini exists, then get_default_path() would return "/etc/zonemaster/backend_config.ini".

If then "/etc/zonemaster/backend_config.ini" is removed, would get_default_path() still return "/etc/zonemaster/backend_config.ini".

Please re-write in simplified language.

Reply:
Is it better now?

Reply:
Yes, thanks.